diff --git a/editor/scene/3d/node_3d_editor_viewport.cpp b/editor/scene/3d/node_3d_editor_viewport.cpp
index 17baadeaf1..f8827f8e6b 100644
--- a/editor/scene/3d/node_3d_editor_viewport.cpp
+++ b/editor/scene/3d/node_3d_editor_viewport.cpp
@@ -5764,7 +5764,7 @@ bool Node3DEditorViewport::_create_instance(Node *p_parent, const String &p_path
 	for (const NodePath &e_path : exposed_nodes) {
 		Node *ei = instantiated_scene->get_node_or_null(e_path);
 		if (ei) {
-			ei->set_meta(META_EXPOSED_IN_INSTANCE, true);
+			ei->set_meta(META_EXPOSED_IN_INSTANCE, instantiated_scene);
 		}
 	}
 
